--- ./setup.py	Sat Apr 27 08:23:32 2013
+++ ./setup.py	Wed Jul 12 15:42:22 2013
@@ -287,19 +287,19 @@
                     span = rconfig_m.span()
                     ok = True
                     break
-                elif rconfig_m is None:
-                    if allow_empty:
-                        print('\nreturned an empty string.\n')
-                        rc += RConfig()
-                        ok = True
-                        break
-                    else:
-                        # if the configuration points to an existing library, 
-                        # use it
-                        if os.path.exists(string):
-                            rc += RConfig(libraries = substring)
-                            ok = True
-                            break
+            if rconfig_m is None:
+               if allow_empty:
+                   print('\nreturned an empty string.\n')
+                   rc += RConfig()
+                   ok = True
+                   break
+               else:
+                   # if the configuration points to an existing library, 
+                   # use it
+                   if os.path.exists(string):
+                       rc += RConfig(libraries = substring)
+                       ok = True
+                       break
             if not ok:
                 raise ValueError('Invalid substring\n' + substring 
                                  + '\nin string\n' + string)
@@ -334,15 +334,30 @@
 
 
 def get_rconfig(r_home, about, allow_empty = False):
+    FORCE_OPTIONS = True #if True, use patch, else use original one
     if sys.platform == "win32" and "64 bit" in sys.version:
         r_exec = os.path.join(r_home, 'bin', 'x64', 'R')
+        r_bin = os.path.join(r_home, 'bin', 'x64')
+        r_include = os.path.join(r_home, 'include')
     else:
         r_exec = os.path.join(r_home, 'bin', 'R')
-    cmd = '"'+r_exec+'" CMD config '+about
-    print(cmd)
-    rp = os.popen(cmd)
-    rconfig = rp.readline()
-    #Twist if 'R RHOME' spits out a warning
+        r_bin = os.path.join(r_home, 'bin', 'i386')
+        r_include = os.path.join(r_home, 'include')
+    options = {}
+    options['--ldflags'] = '-L%s -lR'%(r_bin)
+    options['--cppflags'] = '-I%s'%(r_include)
+    options['LAPACK_LIBS'] = '-lRlapack'
+    options['BLAS_LIBS'] = '-lRblas'
+    if not FORCE_OPTIONS:  
+        cmd = '"'+r_exec+'" CMD config '+about
+        print(cmd)
+        rp = os.popen(cmd)
+        rconfig = rp.readline()
+        #Twist if 'R RHOME' spits out a warning
+        rp.close()
+    else:
+        rconfig = options[about]
+        print "RCONFIG ", about, rconfig
     if rconfig.startswith("WARNING"):
         rconfig = rp.readline()
     rconfig = rconfig.strip()  
@@ -351,7 +364,6 @@
     except ValueError as ve:
         print(ve)
         sys.exit("Problem while running `{0}`\n".format(cmd))
-    rp.close()
     return rc
 
 def getRinterface_ext():